assets/queries/k8s/seccomp_profile_is_not_configured/query.rego

```diff
@@ -1,199 +1,91 @@
-	not common_lib.valid_key(metadata, "annotations")
-
-	result := {
-		"documentId": input.document[i].id,
-		"searchKey": sprintf("metadata.name=%s", [metadata.name]),
-		"issueType": "MissingAttribute",
-		"keyExpectedValue": "'metadata.annotations' is set",
-		"keyActualValue": "'metadata.annotations' is undefined",
-	}
-}
-
-###################################################################
-
-#cronjob
-CxPolicy[result] {
-	document := input.document[i]
-	object.get(document, "kind", "undefined") == "CronJob"
-
-	metadata := document.spec.jobTemplate.spec.template.metadata
-
-	parentMetadata := document.metadata
-	not common_lib.valid_key(metadata, "annotations")
-
-	result := {
-		"documentId": input.document[i].id,
-		"searchKey": sprintf("metadata.name={{%s}}.spec.jobTemplate.spec.template.metadata", [parentMetadata.name]),
-		"issueType": "MissingAttribute",
-		"keyExpectedValue": "'spec.jobTemplate.spec.template.metadata.annotations' is set",
-		"keyActualValue": "'spec.jobTemplate.spec.template.metadata.annotations' is undefined",
-	}
-}
-
-CxPolicy[result] {
-	document := input.document[i]
-	object.get(document, "kind", "undefined") == "CronJob"
-
-	metadata := document.spec.jobTemplate.spec.template.metadata
-
-	parentMetadata := document.metadata
-	common_lib.valid_key(metadata, "annotations")
-
-	annotations := metadata.annotations
-	not common_lib.valid_key(annotations, "seccomp.security.alpha.kubernetes.io/defaultProfileName")
-
-	result := {
-		"documentId": input.document[i].id,
-		"searchKey": sprintf("metadata.name={{%s}}.spec.jobTemplate.spec.template.metadata.annotations", [parentMetadata.name]),
-		"issueType": "MissingAttribute",
-		"keyExpectedValue": "spec.jobTemplate.spec.template.metadata.annotations.seccomp.security.alpha.kubernetes.io/defaultProfileName' is set",
-		"keyActualValue": "'spec.jobTemplate.spec.template.metadata.annotations.seccomp.security.alpha.kubernetes.io/defaultProfileName' is undefined",
+		"keyExpectedValue": sprintf("metadata.name={{%s}}.%s.%s.name={{%s}}.securityContext.seccompProfile.type should be set to 'RuntimeDefault' or 'Localhost'", [metadata.name, specInfo.path, containerType, container.name]),
+		"keyActualValue": sprintf("metadata.name={{%s}}.%s.%s.name={{%s}}.securityContext.seccompProfile.type is set to '%s'", [metadata.name, specInfo.path, containerType, container.name, profile]),
 	}
 }
 
-CxPolicy[result] {
-	document := input.document[i]
-	object.get(document, "kind", "undefined") == "CronJob"
-
-	metadata := document.spec.jobTemplate.spec.template.metadata
-
-	parentMetadata := document.metadata
-	common_lib.valid_key(metadata, "annotations")
-	annotations := metadata.annotations
+# pod defines seccompProfile.type and container inherits this setting
+checkSeccompProfile(specInfo, container, containerType, document, metadata) = result {
+	containerCtx := object.get(container.securityContext, "seccompProfile", {})
+	not common_lib.valid_key(containerCtx, "type")
```

Comment:
Please note that the container check was intentionally formulated to be "optional". Otherwise, the rule would evaluate to _false_ in case `container.securityContext.seccompProfile` is undefined. This could perfectly be the case in practice, as in the subsequent lines, we check for `seccompProfile.type` on pod level.